services/datamanager/builtin/file_deletion.go

```diff
@@ -0,0 +1,135 @@
+package builtin
+
+import (
+	"context"
+	"errors"
+	"io/fs"
+	"math"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"go.viam.com/rdk/logging"
+
+	"github.com/ricochet2200/go-disk-usage/du"
+	"go.viam.com/rdk/services/datamanager/datacapture"
+	"go.viam.com/rdk/services/datamanager/datasync"
+)
+
+// TODO change these values back to what they should be
+const (
+	fsThresholdToTriggerDeletion = .8
+	captureDirToFSUsageRatio     = .5
+	n                            = 4
+)
+
+var errAtSizeThreshold = errors.New("capture dir is at correct size")
+
+func shouldDeleteBasedOnDiskUsage(ctx context.Context, captureDirPath string, logger logging.Logger) (bool, error) {
+	usage := du.NewDiskUsage(captureDirPath)
+	// we get usage this way to ensure we get the amount of remaining space in the partition.
+	// calling usage.Usage() returns the usage of the whole disk, not the user partition
+	usedSpace := 1.0 - float64(usage.Available())/float64(usage.Size())
+	if math.IsNaN(float64(usedSpace)) {
+		return false, nil
+	}
+	if usedSpace < fsThresholdToTriggerDeletion {
+		logger.Warnw("disk not full enough, exiting", "used space", usedSpace)
+		return false, nil
+	}
+	// Walk the dir to get capture stats
+	return exceedsDeletionThreshold(ctx, captureDirPath, float64(usage.Size()), logger)
+}
+
+func exceedsDeletionThreshold(ctx context.Context, captureDirPath string, fsSize float64, logger logging.Logger) (bool, error) {
+	var dirSize int64 = 0
+
+	readSize := func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			if errors.Is(err, fs.ErrNotExist) {
+				return nil
+			}
+			return err
+		}
+		if ctx.Err() != nil {
+			return ctx.Err()
+		}
+
+		if !d.IsDir() {
+			fileInfo, err := d.Info()
+			if err != nil {
+				return err
+			}
+			dirSize += fileInfo.Size()
+			if float64(dirSize)/fsSize > captureDirToFSUsageRatio {
+				logger.Warnw("At threshold to delete, going to delete", "usage ratio", float64(dirSize)/fsSize, "threshold", captureDirToFSUsageRatio)
+				return errAtSizeThreshold
+			}
+		}
+		return nil
+	}
+
+	err := filepath.WalkDir(captureDirPath, readSize)
+	if err != nil && !errors.Is(err, errAtSizeThreshold) {
```

Comment:
Maybe the logic would be clearer with something like like 

```
if err != nil {
    if errors.Is(err, errAtSizeThreshold) return true, nil
    else return false, err
} 
return false, nil
```

Reply:
that is much cleaner, done!